pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go

```diff
@@ -155,7 +155,7 @@ test-e2e: manifests generate fmt vet ## Run the e2e tests. Expected an isolated
 		echo "No Kind cluster is running. Please start a Kind cluster before running the e2e tests."; \
 		exit 1; \
 	}
-	go test ./test/e2e/ -v -ginkgo.v
+	go test ./test/e2e/ -race -v -ginkgo.v
```

Comment:
This slow-down all process for several times and required cgo to be enabled, and on non-Darwin systems requires an installed C compiler. Not sure, that it should be enabled by default.

Probably it can be an option or flag and customer should decide if it enabled or not.
For example, we are using it for long running CI processes running night, but not in regular pipelines for each PR.

Reply:
That sounds great! Thank you a lot for checking it.
So, It seems we have reached a consensus that this should not be added to the default scaffold. 
Users who need it can  always update their own projects accordingly. WDYT? Is that something that make sense based on your investigation?

If so, could you please update the final conclusion in the issue and close both? This way, we can keep track of the reasoning behind not including it.

Thanks!

Reply:
If we want to add some extra flexibility, we can introduce a field that will be empty by default with the ability to change it, and which we will add as arguments when calling tests and/or build.
We can add two different fields for build and tests.
But it is very dangerous to forcefully add the `-race`  flag by default.